docs/understand/texture_fetching.rst

```diff
@@ -0,0 +1,107 @@
+.. meta::
+  :description: This chapter describes the texture fetching modes of the HIP ecosystem
+  ROCm software.
+  :keywords: AMD, ROCm, HIP, Texture, Texture Fetching
+
+*******************************************************************************
+Texture Fetching
+*******************************************************************************
+
+`Textures <https://rocm.docs.amd.com/projects/HIP/en/latest/doxygen/html/group___texture.html>`_ are more than just a buffer, that is interpreted as 1D, 2D or 3D array. Because of its legacy as a graphics functionality, textures are indexed with floats. This can happen two different ways, either the index will be in the range of [0..size-1] or in [0..1]. The difference is mathematically just a division, so for the explanations on this page, we'll use the [0..size-1] indices.
+
+Using float indices isn't trivial. The two issues that come up are sampling and addressing.
+
+When a texture is indexed with a fraction, the queried value is often between two or more texels (texture elements). The sampling method defines what value to return in these cases.
+
+Sometimes the index is outside the bounds of the texture. This might seem unintuitive, but is useful in computer graphics for putting a texture on a surface multiple times, or just creating a visible sign of out of bounds indexing. Addressing mode defines what value to return when indexing a texture out of bounds.
+
+There are several different sampling and addressing modes, and no rule for how to choose between them, as it is dependent on the application. This means that it has to be decided during development which sampling and addressing method to use.
+
+This image is the texture, that will be used as for the examples below. It is 2x2 texels and is indexed in the [0..1] range.
+
+.. image:: ../data/understand/textures/original.png
+  :width: 150
+  :alt: Texture used as example
+
+
+.. _texture_fetching_nearest:
+Nearest point sampling
+===============================================================================
+
+In this mode the ``tex(x) = T[floor(x)]`` and similarly for 2D and 3D variants. In practice this will not interpolate between neighboring values, which results in a pixelated look.
+
+This image is the example texture stretched out to a 4x4 pixel quad, but still indexed in the [0..1] range. The in between values are the same as the values of the nearest texel.
```

Comment:
This is why I would want some kind of coordinate-grid on top of this: how is this "nearest point" sampling, when it just rounds the coordinate down? I know, the name is probably just convention by now.

But if you index this in the [0...1] range despite stretching it out to a 4x4 pixel, how does this create this picture? Assuming that the upper left is (0,0), and every pixel adds 0.25 to the indexing, then the third pixel to the right in the first row would have the index (0, 0.75), which by the formula for this sampling would index the original texture at (0,0), thereby being red. Wouldn't 3/4th of this sampling method have to be red?

Reply:
the sample points should be 0, 0.33, 0.66 and 1, on both axis